### DISCUSSION TITLE
Is there a list of borg commands that do not need the passphrase?

### DISCUSSION THREAD

**Discussion by kmille** [borgbackup/borg#8015] [Q&A] [answered]
Is there a list of borg commands that do not need the passphrase? Some need the passphrase, some do not. But which ones?
I'm asking for a borg wrapper I'm currently working on.

**Comment by ThomasWaldmann** [comment]
Commands dealing with repository layer only will not need the passphrase.

Everything that deals also with archives / list of archives will need the passphrase.

For a wrapper, I'ld just always have the passphrase readily available in `BORG_PASSPHRASE` env var when invoking borg.

**Comment by kmille** [comment]
I'm currently working on [borgctl](https://github.com/kmille/borgctl).
> For a wrapper, I'ld just always have the passphrase readily available in BORG_PASSPHRASE env var when invoking borg.

The thing is: I don't want to store my passphrase on disk. If `ask` is specified in the borgctl config file, I ask the user to enter the password. This would not be needed if the password is not needed. Especially prune/compact/check are the critical/important/most used commands.

**Reply by ThomasWaldmann** [reply, marked as answer]
passphrase not needed: `compact`, `borg check --repository-only` (IIRC)

**Reply by kmille** [reply]
also `break-lock` and `umount` does not the passphrase
